service/worker/scheduler/workflow.go

```diff
@@ -1424,3 +1422,11 @@ func GetListInfoFromStartArgs(args *schedspb.StartScheduleArgs, now time.Time, s
 	s.State.LastProcessedTime = timestamppb.New(now)
 	return s.getListInfo(false)
 }
+
+func InternalWorkflowIdRepresentation(originalWorkflowId string, startTime time.Time, keepOriginalWorkflowId bool) string {
+	timeSec := startTime.UTC().Truncate(time.Second)
+	if keepOriginalWorkflowId {
+		return originalWorkflowId
```

Comment:
When this option is activated, does that mean there is a higher chance of having the same ID and therefore more "WorkflowAlreadyStartedErr" cases (depending on the schedules and whether they overlap)? If so, is that a concern?